spec/core/scheduler_spec.rb

```diff
@@ -98,12 +98,6 @@
         it_behaves_like 'does not trigger', :sync
       end
 
-      context 'when connector is never synced' do
-        let(:last_synced) { nil }
-
-        it_behaves_like 'triggers', :sync
-      end
-
```

Comment:
I think it's nice to have a test for this case - if connector never synced and schedule says "every hour" - when does the trigger happen if it's 10:30? If it's 10:55? If it's 10:02?

From looking at the code I can't answer this question and it would be great to express answer to this question with a test/couple tests.

Reply:
When `last_synced = nil`, we will default it to now, and use it to find the next run, which should always be in the future.

But in the test, I don't want to test `Fugit::Cron#next_time`, instead I only want to mock it. 

In this case, what I can test is, when `last_synced = nil`, and mock `Fugit::Cron#next_time` to return a future date, the expected result is it will not trigger a job. This is the same as this example:
https://github.com/elastic/connectors-ruby/blob/59f526ff017e38da596851800bc7f4508f328552/spec/core/scheduler_spec.rb#L121-L125

That's why I didn't add an example.

Reply:
This is what I don't honestly understand :) 

We have a code that triggers sync when the time comes, which is:

```
next_trigger_time = cron_parser.next_time(last_synced)

# Sync if next trigger for the connector is in past
if next_trigger_time < Time.now
  # do sync
end
```

When `last_synced` will be Time.now, then this code will never trigger, how will the scheduled trigger happen?

Reply:
Hey @artem-shelkovnikov , this is a really good point.